notebooks/API.py

```
@@ -0,0 +1,189 @@
+@app.post("/predict")
+def predict(transaction: TransactionInput):
+    try:
+        # Convert input to DataFrame
+        input_dict = transaction.dict()
+        df = pd.DataFrame([input_dict])
+
+        # Apply business rules
+        df = apply_business_rules(df)
+
+        # Preprocess the data
+        processed_data = preprocess_test(df, scaler, encoder)
```

Ruff (0.8.2)

Undefined name `preprocess_test`

(F821)